From 09f10c1b190f93b971b3a2746cf1dfc7467eb3bb Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Mon, 19 Nov 2018 09:58:32 +0200
Subject: [PATCH 0905/1200] phy: marvell cp110: mark u-boot power-off calls

It helps ATF to determine who called power off
function (U-boot/Linux) and act accordingly

Change-Id: I60dd34507502397ac5b37f7dfc1c1d4251ccb10f
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61035
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Grzegorz Jaszczyk <jaz@semihalf.com>
---
 drivers/phy/marvell/comphy_cp110.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/phy/marvell/comphy_cp110.c b/drivers/phy/marvell/comphy_cp110.c
index d1a33676be..d64711f0f8 100644
--- a/drivers/phy/marvell/comphy_cp110.c
+++ b/drivers/phy/marvell/comphy_cp110.c
@@ -551,10 +551,10 @@ int comphy_cp110_init(struct chip_serdes_phy_config *ptr_chip_cfg,
 		}
 		switch (ptr_comphy_map->type) {
 		case COMPHY_TYPE_UNCONNECTED:
+			mode = COMPHY_TYPE_UNCONNECTED | COMPHY_CALLER_UBOOT;
 			ret = comphy_smc(MV_SIP_COMPHY_POWER_OFF,
 					 ptr_chip_cfg->comphy_base_addr,
-					 lane,
-					 ptr_comphy_map->type);
+					 lane, mode);
 		case COMPHY_TYPE_IGNORE:
 			continue;
 			break;
-- 
2.22.0

